tests/utils.py

```diff
@@ -397,10 +492,17 @@ def compare_all_settings(model: str,
 
             if method == "generate":
                 results += _test_completion(client, model, prompt, token_ids)
+            elif method == "generate_close":
+                results += _test_completion_close(client, model, prompt)
+            elif method == "generate_with_image":
+                results += _test_image_text(
+                    client, model,
+                    "https://upload.wikimedia.org/wikipedia/commons/0/0b/RGBA_comp.png"
+                )
             elif method == "encode":
                 results += _test_embeddings(client, model, prompt)
             else:
-                assert_never(method)
+                raise ValueError(f"Unknown method: {method}")
```

Comment:
`assert_never` also throws an error when reached. The advantage of using `assert_never` is that it fails the type checker as well (not currently enforced in CI but you can see it in the IDE). Why not update the type definition for `method`?

Reply:
because I need to pass string from the caller, and the type checker will complain if it is string literal.